test/rules-adobe-fullstory.spec.ts

```diff
@@ -1,56 +1,94 @@
+      let testHarness: RulesetTestHarness;
 
-    ExpectObserver.getInstance().create(
-      { rules: [expectRule('fs-event-adobe-evars')], readOnLoad: true },
-    );
+      afterEach(async () => {
+        await testHarness.tearDown();
+      });
 
-    expectNoCalls(expectGlobal('FS'), 'event');
-  });
+      after(async () => {
+        await testEnv.tearDown();
+      });
+
+      it('sends Adobe eVars and props to FS.event', async () => {
+        testHarness = await testEnv.createTestHarness(adobeRules, {
+          s: {
+            eVar1: '',
+            eVar10: '',
+            eVar20: '',
+            eVar50: '',
+            eVar60: '',
+            prop1: '',
+            prop10: '',
+            prop20: '',
+            prop50: '',
+            prop60: '',
+            pageName: '',
+          } as AppMeasurement,
+        });
+
+        await testHarness.execute(([localAppMeasurement]) => {
+          globalThis.s.eVar1 = localAppMeasurement.eVar1;
+          globalThis.s.eVar10 = localAppMeasurement.eVar10;
+          globalThis.s.eVar20 = localAppMeasurement.eVar20;
+          globalThis.s.eVar50 = localAppMeasurement.eVar50;
+          globalThis.s.eVar60 = localAppMeasurement.eVar60;
+
+          globalThis.s.prop1 = localAppMeasurement.prop1;
+          globalThis.s.prop10 = localAppMeasurement.prop10;
+          globalThis.s.prop20 = localAppMeasurement.prop20;
+          globalThis.s.prop50 = localAppMeasurement.prop50;
+          globalThis.s.prop60 = localAppMeasurement.prop60;
+
+          globalThis.s.pageName = localAppMeasurement.pageName;
+        }, [basicAppMeasurement]);
+
+        let [eventName, payload] = await testHarness.popEvent();
+        expectEqual(eventName, 'props');
+        expectMatch(basicAppMeasurement, payload, 'prop1', 'prop10', 'prop20', 'prop50', 'prop60');
+        expectUndefined(payload, 'pageName');
+
+        [eventName, payload] = await testHarness.popEvent();
+        expectEqual(eventName, 'eVars');
+        expectMatch(basicAppMeasurement, payload, 'eVar1', 'eVar10', 'eVar20', 'eVar50', 'eVar60');
+        expectUndefined(payload, 'pageName');
+      });
+
+      it('does not send empty eVar or prop objects to FS.event', async () => {
+        testHarness = await testEnv.createTestHarness(adobeRules, { s: basicAppMeasurement });
 
-  it('it should FS.identify using specific Adobe eVars', () => {
-    ExpectObserver.getInstance().create(
-      { rules: [expectRule('fs-identify-adobe-evars')], readOnLoad: true },
-    );
+        await testHarness.execute(() => {
+          globalThis.s.eVar1 = undefined;
+          globalThis.s.prop1 = undefined;
+        });
 
-    const [uid, payload] = expectFS('identify');
-    expectEqual(uid, basicAppMeasurement.eVar1);
-    expectMatch(basicAppMeasurement, payload, 'eVar10', 'eVar20');
-    expectUndefined(payload, 'prop1', 'pageName');
+        await new Promise<void>((resolve, reject) => {
```

Comment:
I'm repeating this logic a lot. I think I'd like to roll out a general "cleanup" PR after getting all the ruleset tests converted (just Tealium left after this). In addition to extracting this method, I have an idea for cleaning up test env and harness initialization and to potentially reuse browser instances across tests for different rulesets which should shorten browser test run time.